pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -472,17 +480,26 @@ def save_checkpoint(self, checkpoint: Dict, filepath: str) -> None:
         """Save model/training states as a checkpoint file through state-dump and file-write.
 
         Args:
+            checkpoint: The checkpoint state dictionary
             filepath: write-target file's path
-            weights_only: saving model weights only
         """
         if self.world_size > 1 and self.zero_stage_3:
+            save_dir = self._filepath_to_dir(filepath)
+            if self.save_full_weights:
+                # todo: expose this as a unprotected function in deepspeed
+                state_dict = self.deepspeed_engine._zero3_consolidated_fp16_state_dict()
+                if self.is_global_zero:
+                    # Remove module portion of the checkpoint before saving
```

Comment:
Can you explain what is a module portion in the comment ?

Reply:
is this intending to not save the full weights on ranks other than rank 0? if so, does the check need to be if the self.global_rank > 1 instead of self.world_size > 1 ?

Reply:
I shamelessly took the code from https://discuss.pytorch.org/t/prefix-parameter-names-in-saved-model-if-trained-by-multi-gpu/494/2

I'll clear it up in the comment, but recall we always wrap our model in a wrapper class (`LightningDeepSpeedModule`). This means the state dict includes this as a prefix across the keys. I delete this prefix before saving on the keys :)